src/frontend/src/optimizer/rule/trivial_project_to_values_rule.rs

```diff
@@ -0,0 +1,42 @@
+// Copyright 2023 RisingWave Labs
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use super::{BoxedRule, Rule};
+use crate::optimizer::plan_node::LogicalValues;
+use crate::optimizer::PlanRef;
+
+pub struct TrivialProjectToValuesRule {}
+impl Rule for TrivialProjectToValuesRule {
+    fn apply(&self, plan: PlanRef) -> Option<PlanRef> {
+        let project = plan.as_logical_project()?;
+        if project.exprs().iter().all(|e| e.is_const()) {
+            Some(
+                LogicalValues::new(
+                    vec![project.exprs().clone()],
+                    project.schema().clone(),
+                    project.ctx(),
+                )
+                .into(),
+            )
+        } else {
+            None
+        }
```

Comment:
We need to ensure project's input produces exact one row.

Reply:
Hmm, true, else the number of rows can be dynamic...